src/query.rs

```diff
@@ -762,6 +817,191 @@ macro_rules! tuple_impl {
 //smaller_tuples_too!(tuple_impl, B, A);
 smaller_tuples_too!(tuple_impl, O, N, M, L, K, J, I, H, G, F, E, D, C, B, A);
 
+/// A prepared query can be stored independently of the [`World`] to amortize query set-up costs.
+pub struct PreparedQuery<Q: Query> {
+    memo: (u64, u64),
+    state: Box<[(usize, <Q::Fetch as Fetch<'static>>::State)]>,
+}
+
+impl<Q: Query> Default for PreparedQuery<Q> {
+    fn default() -> Self {
+        Self {
+            // This memo will not match any world as the first ID will 1.
+            memo: (0, 0),
+            state: Default::default(),
+        }
+    }
+}
+
+impl<Q: Query> PreparedQuery<Q> {
+    #[cold]
+    fn new(world: &World) -> Self {
```

Comment:
Is this dead code? I didn't see any calls.

Reply:
It is called in `PreparedQuery::query` and `::query_mut`:
```rust
if self.memo != world.memo() {
    *self = Self::new(world);
}
```
but I will rename it to `prepare` and expose a public `new` that is equivalent to the `Default` impl.

Reply:
Ah, that's why my grep didn't find it. SGTM.